microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,221 @@
+        "build-essential",
+        "libnuma-dev",
+        "libmnl-dev",
+        "libelf-dev",
+        "dpkg-dev",
+        "pkg-config",
+        "python3-pip",
+        "python3-pyelftools",
+        "python-pyelftools",
+    ]
+    _redhat_packages = [
+        "gcc",
+        "make",
+        "git",
+        "tar",
+        "wget",
+        "dos2unix",
+        "psmisc",
+        "kernel-devel",
+        "numactl-devel.x86_64",
+        "librdmacm-devel",
+        "pkgconfig",
+        "libmnl-devel",
+        "elfutils-libelf-devel",
+        "python3-pip",
+    ]
+    _rte_target = "x86_64-native-linuxapp-gcc"
+    _dpdk_github = "https://github.com/DPDK/dpdk.git"
+    _ninja_url = (
+        "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip"
+    )
+
+    _nic_attributes_regex = r"([a-zA-Z_]+),?"
+
+    def _install_dpdk_dependencies(self, node: Node) -> None:
+        if isinstance(node.os, Ubuntu):
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
+            self.log.info("Packages installed for Ubunutu")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "mv /usr/bin/meson /usr/bin/meson.bak")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+            self._execute_expect_zero(node, "pip3 install --upgrade ninja")
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self._execute_expect_zero(
+                node, "yum update -y --disablerepo='*' --enablerepo='*microsoft*'"
+            )
+            node.os.install_packages(
+                ["groupinstall", "'Infiniband Support'"], signed=False
+            )  # todo gross hack to support groupinstall
+            for package in self._redhat_packages:
+                node.os.install_packages(package)
+            result = node.execute(
+                "dracut --add-drivers 'mlx4_en mlx4_ib mlx5_ib' -f"
+            )  # add mellanox drivers
+            self.log.debug("\n".join([result.stdout, result.stderr]))
+            self._execute_expect_zero(node, "systemctl enable rdma")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+
+            self._execute_expect_zero(
+                node,
+                f"wget {self._ninja_url}",
+            )
+            self._execute_expect_zero(
+                node, "unzip ninja-linux.zip && mv ninja /usr/bin/ninja"
+            )
+            self._execute_expect_zero(node, "pip3 install --upgrade pyelftools")
+
+    def _execute_expect_zero_with_path(
+        self, node: Node, cmd: str, path: PurePath
+    ) -> str:
+        result = node.execute(cmd, sudo=True, cwd=path, shell=True)
+        result.assert_exit_code()
+        self.log.info(result.stdout)  # TODO: debug
```

Comment:
Yeah, not a good idea to have this at info level (which is default triggered)

Reply:
BTW, you've packed a bunch of things in that one commit, why split this random small part out of that? Just squash and be happy.

Reply:
Thank you! I will tweak the info levels and squash before the final merge, I'm working on this one and the PR is just a rolling review before the final one 🙂